### PR TITLE
Misc fixes doc changes

### PR DESCRIPTION
**Pull request by wmamills**
Multiple small doc changes as a group.
If you wish to veto one I will rebase it out.

### CODE REVIEW

**Review by arnopo — APPROVED** [review]
LGTM

File: openamp/contributing.rst [hunk 1]
```diff
@@ -34,6 +34,6 @@ Platform maintainers
 
 Push rights
 -----------
-- Push rights restricted to Core Team
-- Generally exercised by top-level maintainer (Wendy Liang)
-- Maintainer manages delegation within core team
+- Push rights restricted to the Core Team
+- Generally exercised by the maintainers for each repository
+- Maintainers manages delegation between themselves
```
Comment (tammyleino) [inline openamp/contributing.rst:39]: Maintainers "manage"

**Comment by tammyleino** [comment]
Approved with minor comment.

**Review by tammyleino — APPROVED** [review]

File: openamp/contributing.rst (hunk 1 above)
Comment (wmamills) [reply openamp/contributing.rst]: Pushed again to fix this issue.

**Pull request merged**